src/yajra/Datatables/Engines/BaseEngine.php

```diff
@@ -667,7 +667,20 @@ public function render($object = false)
 
         if (isset($this->transformer)) {
             $fractal        = new Manager();
-            $resource       = new Collection($data, new $this->transformer());
+            
+            //Get transformer reflection
+            //Firs method parameter should be data/object to transform
+            $reflection = new \ReflectionMethod($this->transformer,'transform');
+            $parameter = $reflection->getParameters()[0];
+
+            //If parameter is class assuming it requires object
+            //Else just pass array by default
+            if($parameter->getClass()){
+                $resource = new Collection($this->result_object, new $this->transformer());
+            }else{
+                $resource = new Collection($this->result_array_r, new $this->transformer());
```

Comment:
Maybe we should use the original code here to avoid potential breaking change?
`$resource       = new Collection($data, new $this->transformer());`